lib/std/os/linux.zig

```diff
@@ -860,6 +860,20 @@ pub fn fstatat(dirfd: i32, path: [*]const u8, stat_buf: *Stat, flags: u32) usize
     }
 }
 
+pub fn statx(dirfd: i32, path: [*]const u8, flags: u32, mask: u32, statx_buf: *Statx) usize {
+    if (@hasDecl(@This(), "SYS_statx")) {
+        return syscall5(
+            SYS_statx,
+            @bitCast(usize, isize(dirfd)),
+            @ptrToInt(path),
+            flags,
+            mask,
+            @ptrToInt(statx_buf),
+        );
+    }
+    return @bitCast(usize, isize(-ENOSYS));
```

Comment:
Would it be better to just let the function fail compilation at compile time?

Reply:
I don't think so, from a user point of view you'd have to add some comptime logic in order to avoid calling `statx` and you'd have to also provide a fallback path to handle the `ENOSYS` case anyway.